### DISCUSSION TITLE
How do I install this on my host?

### DISCUSSION THREAD

**Discussion by JesusGod-Pope666-Info** [mmontag/chip-player-js#168] [Q&A]
How do I install this with my own music browse folder?
Is there an easy way to just copy some files to a host domain and have it working quick?

Any way to change the Midi fonts, I think they are called?

I looked in the readme file and was.... Okay, I'll just ask! :P

-- Darkijah

**Comment by JesusGod-Pope666-Info** [comment]
Could you release an easy install for ones own website, with a folder one can put our files into very easy peasy to get to work :)

And if MP3 is not working, well, could be pretty nice with some support for that as well ;) AND MP4 :) Just to play the sound.

It is very nice with that old design of yours. I would like to see it further improved. And still that Visualizer is just awesome.

**Reply by mmontag** [reply]
Yeah, the self hosting process could be improved, as it involves a ton of work setting up Emscripten, building emulation libraries, etc. 

Easy way to improve this would be to provide instructions for pulling in a pre-built chip-core.wasm (or in the worst case, track this in the repository). This way, people won't have to run `build-chip-core` themselves if they only want to self-host or tweak the frontend part of the app. 

The wasm bundle could even be forked into its own standalone project, which some people have also asked for, but I'm getting ahead of myself.

**Reply by JesusGod-Pope666-Info** [reply]
Yea I just don't have much of a idea how to host it on my website - to fill it up with my own music folders and such.
But I really like the Visualizer, had to say it again. You did really really good with that! Just a joy to look at the music being played and seeing what is going on and getting it.

I don't know.... Would be nice with an easy quick setup for self hosting for sure that newbs Like me could easily copy past and be one the fly to fill it with out own music.

Of cause, the Visualizer could probably be tuned with further color themes maybe for further fun stuff to it. And still would like to see it further have space in the player like described in another place. Like it is such a nice job and if one could gain even more space for it, it would be amazing. I am also trying things here and there and the moment - this wav thing, where it has future, present and past waves, I think they call it a waterfall effect or something....

-- Darkijah

**Reply by blsaws** [reply]
I'm very interested in getting self-hosting to be a viable option for this app. The gh-pages branch seems to be a mostly viable baseline for deployment as-is, if only I could figure out how to setup the catalog of files/directories, so that the app discovers it where it's supposed to be on startup. The only reference I've seen so far is in the README re "The music catalog is created by..." which is part of the build process apparently. But my experience with the build process ran into numerous issues, that I would be happy to share / help resolve (and even fully automate the process if possible) if there's someone to work with on that, so I can get more design details e.g. internal app arch and data flow. As a start, if possible please explain if there's a way to stage/plugin files and file metadata for a run-ready deployment. Thanks for any guidance.
